env/env.zsh

```diff
@@ -0,0 +1,5 @@
+export PATH=$HOME/.dune/bin:$PATH
+autoload -Uz compinit bashcompinit
+compinit
+bashcompinit
+source $HOME/.dune/completions/bash.sh
```

Comment:
zsh can do that? take a bash completion script and just work with it directly?
wow

Reply:
Yep! If there's a similar thing to do with `fish` we could add it to the fish env too. It looks like there are a few ways to get fish to understand a bash completion script. As a fish user, is there an obvious choice for how to do it?

Reply:
Even if there's no obvious choice, maybe it would make sense to provide an example of how a fish user could take advantage of the bash completion script. Like the install script could print an example fish config. I won't include that in this PR but you're welcome to add it in a later change if it's something you're interested in.